validatedeps.js

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env node
+/**
+ * Created by cmeyers on 9/20/16.
+ */
+const fs = require('fs');
+const start = new Date().getTime();
+
+const args = process.argv.slice(2);
+
+var projectPath = '';
+
+if (args.length === 1) {
+    const pathParts = args[0].split('=');
+
+    if (pathParts.length !== 2) {
+        console.error('you must specify a project path in the form -p=path/to/project');
+        process.exit(1);
+    }
+
+    projectPath = pathParts.slice(-1);
+}
+
+const workingPath = process.cwd();
+const resolvedPath = buildPath(`${workingPath}/${projectPath}`);
+console.log(`validating dependencies in ${resolvedPath}`);
+const packageJsonPath = buildPath(`${resolvedPath}/package.json`);
+const shrinkwrapJsonPath = buildPath(`${resolvedPath}/npm-shrinkwrap.json`);
+
+// match x, tilde, gt, lt, star, carat or whitespace in version
+const IMPRECISE_VERSION_CHARS_PATTERN = /[x~><*|\^\s]+/;
```

Comment:
@tfennelly I wanted to keep this as a standalone script; using the `semver` package would require a package.json and an install process. Trying to avoid making this more complex than it needs to be.

@sophistifunk that pattern would fail the "-beta1" version format we use for `@jenkins-cd` libs which would cause our PR branch builds to fail. I can certainly write a regex for that pattern too.